### ISSUE TITLE
Cron job 'jetpack_v2_heartbeat' not removed after Jetpack is deactivated

### ISSUE THREAD

**Issue by lakellie** [Automattic/jetpack#30382]
### Impacted plugin

Jetpack

### Quick summary

A Jetpack user in the forums has reported that the cron job `jetpack_v2_heartbeat` is not removed after deactivation: https://wordpress.org/support/topic/wp-cron-jetpack_v2_heartbeat-not-being-removed/

Can y'all confirm whether this needs to be added to the list of things that are removed upon plugin deactivation/deletion (or if it already is included and there's perhaps something else going on here?)

Thanks!

**Comment by jeherve** [comment]
> Can y'all confirm whether this needs to be added to the list of things that are removed upon plugin deactivation/deletion (or if it already is included and there's perhaps something else going on here?)

Yes, that's something we could add to the uninstall routine for sure, or / and to the disconnection method:

https://github.com/Automattic/jetpack/blob/210e75345653abee7954a11d3669cd56175b9ea5/projects/packages/connection/src/class-manager.php#L2000-L2007

cc @Automattic/jetpack-kernel, who may have more ideas there.

**Comment by jeherve** [comment]
Somewhat related:

- #30243 
- #28540 

It may be nice to do an audit of what needs to happen when Jetpack is deactivated when other Jetpack plugins are still vs. not, and ensure that we only leave what's necessary based on the situation.

**Comment by bjoernzosel** [comment]
I have also 2 cron jobs that I can not delete, although I deinstalled the plugin and cleaned the database:

jetpack_clean_nonces
jetpack_v2_heartbeat

**Comment by superbmario** [comment]
I have the same issue, two cron jobs still active after Jetpack was uninstalled:

jetpack_clean_nonces
jetpack_v2_heartbeat

Update: It looks like they're both present in WooCommerce. e.g. 'jetpack_v2_heartbeat' is identified in `woocommerce/vendor/automattic/jetpack-connection/src/class-heartbeat.php` file.

Does anyone know it can be safely removed?

**Comment by zaerl** [comment]
I will take care of this.
